internal/pagination/books.go

```diff
@@ -0,0 +1,172 @@
+package pagination
+
+import (
+	"bookshop/internal/models"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const (
+	formatParam   = "format"
+	categoryParam = "category"
+	minPriceParam = "min_price"
+	maxPriceParam = "max_price"
+)
+
+type BookPaginationData struct {
+	*PaginationData
+	Format   []string
+	Category []int64
+	MinPrice float64
+	MaxPrice float64
+}
+
+func ParseBookPaginationQuery(r *http.Request) (*BookPaginationData, error) {
+	qs := r.URL.Query()
+
+	p, err := ParsePaginationData(r, bookItemSortKeyValidator)
+	if err != nil {
+		return nil, err
+	}
+
+	bp := &BookPaginationData{PaginationData: p}
+
+	if formats, err := parseAndValidateFormats(qs.Get(formatParam)); err != nil {
+		return nil, err
+	} else if len(formats) > 0 {
+		bp.Format = formats
+	} else {
+		bp.Format = []string{string(models.Paperback)}
+	}
+
+	if categories, err := parseCategories(qs.Get(categoryParam)); err != nil {
+		return nil, err
+	} else {
+		bp.Category = categories
+	}
+
+	if minPrice, maxPrice, err := parseMinMaxPrice(qs.Get(minPriceParam), qs.Get(maxPriceParam)); err != nil {
```

Comment:
```
minPrice, maxPrice, err := parseMinMaxPrice(qs.Get(minPriceParam), qs.Get(maxPriceParam))
	if err != nil {
		return nil, err
	}
	bp.MinPrice = minPrice
	bp.MaxPrice = maxPrice
	```
	
Can we check on errors without else blocks? Why? It won't shadow the err variable created previously. And it would be more readable

Reply:
fixed